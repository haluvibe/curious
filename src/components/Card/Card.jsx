import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "material-ui/styles";
import Card, { CardContent } from "material-ui/Card";
import Button from "material-ui/Button";
import Typography from "material-ui/Typography";

const styles = theme => ({
  card: {
    minWidth: 275
  },
  bullet: {
    display: "inline-block",
    margin: "0 2px",
    transform: "scale(0.8)"
  },
  title: {
    marginBottom: 16,
    fontSize: 14,
    color: theme.palette.text.secondary
  },
  pos: {
    marginBottom: 12,
    color: theme.palette.text.secondary
  }
});

function SimpleCard (props) {
  const { classes, children, title, headline } = props;

  return (
    <div>
      <Card className={classes.card}>
        <CardContent>
          {title && <Typography className={classes.title}>{title}</Typography>}

          {headline && 
            <Typography variant="headline" component="h2">
              {headline}
            </Typography>
          }

          {children}
        </CardContent>
      </Card>
    </div>
  );
}

SimpleCard.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(SimpleCard);
