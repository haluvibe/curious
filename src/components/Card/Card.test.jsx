import React from "react";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";

import Card from "./Card";

test("Card component should render as expected", () => {
  const component = shallow(<Card />),
    tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
