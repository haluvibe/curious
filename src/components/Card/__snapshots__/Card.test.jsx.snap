// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render as expected 1`] = `
<SimpleCard
  classes={
    Object {
      "bullet": "SimpleCard-bullet-2",
      "card": "SimpleCard-card-1",
      "pos": "SimpleCard-pos-4",
      "title": "SimpleCard-title-3",
    }
  }
/>
`;
